### ISSUE TITLE
Monsters only regen on current map

### ISSUE THREAD

**Issue by GoogleCodeExporter** [alistairreilly/andors-trail#111]
```
What steps will reproduce the problem?
So this is not a major issue but the only monsters that regenerate are the ones 
on the current map that you are on. The other maps that I walk to are pretty 
much empty since I generally lay waste to all of the monsters on all given maps.


What is the expected output? What do you see instead?
I would love to have the monsters on other maps regenerate maybe slower than 
the current map that you are on and maybe all at once if that makes the 
processing easier in the game.

What version of the product are you using? On what operating system?
Version 0.6.7 on android 2.2 for HTC Droid Incredible



```

Original issue reported on code.google.com by `dextermi...@gmail.com` on 28 Dec 2010 at 8:27

**Comment by GoogleCodeExporter** [comment]
```
Thanks for the report. 

Yes, I totally agree. Only the current map is now subject to respawning. This 
is something that I want to change. If you haven't visited a map in N 
seconds/minutes, the map should completely respawn.

Would 5 minutes be a good time limit to when we respawn maps other than the 
current one?
```

Original comment by `oskar.wi...@gmail.com` on 9 Jan 2011 at 7:44
- Changed state: **Accepted**
- Added labels: **Milestone-0.6.8**

**Comment by GoogleCodeExporter** [comment]
```
Five minutes sounds good to me. Thanks for the response. Great game by the
way. I look forward to then next update for more maps and quests.

map

This is something that I want to change. If you haven't visited a map in N
seconds/minutes, the map should completely respawn.

the current one?
```

Original comment by `dextermi...@gmail.com` on 9 Jan 2011 at 7:52

**Comment by GoogleCodeExporter** [comment]
```
The respawning on the current map will remain the way it is now.

Other maps than the current one will respawn every N minutes.
```

Original comment by `oskar.wi...@gmail.com` on 9 Jan 2011 at 8:23

**Comment by GoogleCodeExporter** [comment]
```
This is now implemented and will be included in the next version. I changed the 
respawning time to 3 minutes. So if you haven't visited a map in 3 minutes 
(real-world time), all non-unique monsters will now respawn. This also includes 
time spent not actually playing the game, so if you load an old savegame, the 
other maps than the current one will probably have respawned.

Thank you dextermichael2003 for the good idea! Please report more things that 
we can add to improve the game.
```

Original comment by `oskar.wi...@gmail.com` on 9 Jan 2011 at 10:50
- Changed state: **Ready**
- Added labels: **Type-Enhancement**
- Removed labels: **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
```
Hm... 

1. How about adding this last part (real time outside the game) as an option? I 
understand that it's quite hard with android's stateful concept (press Home 
button and application is actually still running).

2. This kinda conflicts with the turn-based concept. How about counting "turns" 
e.g. 1 combat turn = 1, 1 step = 1 or maybe 2. 8-hour rest = 4800 turns. 
Respawn occurs if map was abandoned for, say, 100 turns. (hey, I may have 
invented in-game TIME ;) )

3. Rest means "respawn all", doesn't it.

3. How about entering and then leaving a map every 2 minutes? will it never 
respawn?
e.g. I kill forest snakes south of village, then two more snakes south of that, 
then return to village to sell meat / buy minor healing.
I spend
- 1 minute in village
- 1 minute in wilderness1 (w1)
- 1 minute in wilderness2 (w2)
- 10 seconds crossing w1 towards village (no respawn yet)
- 1 minute in village again
- 10 seconds in w1 (still no respawn)
- 10 seconds in w2 (no respawn)...


```

Original comment by `surrano` on 9 Jan 2011 at 3:03

**Comment by GoogleCodeExporter** [comment]
```
If you not moving while in-game and doing nothing (checking inv., using items, 
moving etc is an action AND game is open, HOME button should effectively 
"pause" everything) should speed up the respawn rate. Old respawn rate was 
fine, it just needs the boost when idle.
This will allow you to clear maps and continue onto next map without having to 
re-kill monsters on the past cleared maps.
```

Original comment by `kims...@gmail.com` on 9 Jan 2011 at 3:11

**Comment by GoogleCodeExporter** [comment]
```
I also have a problem with the realtime approach. Did you only use it because 
of simplicity? (I could understand this, its not as easy as it sounds)

But the main problem would be in dungeons:
I would have to clear hard dungeons really fast to get back alive. (without too 
many monsters revived - would almost double the hardness) Perhaps suspend that 
rule if you are still in the dungeon.
```

Original comment by `SamuelPl...@gmail.com` on 13 Jan 2011 at 10:10

**Comment by GoogleCodeExporter** [comment]
```
Good point. Although, I think that making it that much harder is good. I know I 
look forward to snakes re-spawning so i can get them on the way out. Part of 
going in means you have to have a exit strategy too. We don't want the game to 
be too easy. I'm all for a challenge.
```

Original comment by `Captain8...@gmail.com` on 14 Jan 2011 at 1:19

**Comment by GoogleCodeExporter** [comment]
```
Released in v0.6.8 . Closing.
```

Original comment by `oskar.wi...@gmail.com` on 24 Feb 2011 at 9:35
- Changed state: **Fixed**
